tools/internal_ci/linux/pull_request/grpc_build_artifacts.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# Copyright 2021 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+# change to grpc repo root
+cd $(dirname $0)/../../../..
+
+source tools/internal_ci/helper_scripts/prepare_build_linux_rc
+
+set +ex
+[[ -s /etc/profile.d/rvm.sh ]] && . /etc/profile.d/rvm.sh
+set -e  # rvm commands are very verbose
+rvm --default use ruby-2.4.1
+set -ex
+
+# Linux artifacts includes C#, Python, Ruby, PHP, Protoc. Python, Ruby, PHP
+# artifacts are being built inside individual distribtest jobs. Core-based C# is
+# in sunset mode. So, we only build protoc here to avoid duplicated works.
+tools/run_tests/task_runner.py -f artifact linux presubmit protoc -j 6
```

Comment:
only building protoc artifacts here is quite controversial and it has some problems:
- C# artifacts won't be tested on PRs at all
- ruby distribtests that run on PRs aren't 100% identical to the distribtests that run in the build flow (there's an issue with building the ruby grpc-tools gem).
- potentially other problems

Let's leave out this change for now and look into introducing it in a separate PR (as its orthogonal to what this PR is trying to do).

Reply:
Sounds good. Reverted for now.